go/processtree/slavenode.go

```diff
@@ -66,92 +62,85 @@ const (
 func (tree *ProcessTree) NewSlaveNode(identifier string, parent *SlaveNode, monitor filemonitor.FileMonitor) *SlaveNode {
 	s := SlaveNode{}
 	s.needsRestart = make(chan bool, 1)
-	s.commandBootRequests = make(chan *CommandRequest, 256)
 	s.slaveBootRequests = make(chan *SlaveNode, 256)
-	s.parentReadiness = make(chan bool, 1)
-	s.childBootRequests = make(chan *SlaveNode, 256)
-	s.Features = make(map[string]bool)
-	s.event = make(chan bool, 1)
+	s.commandBootRequests = make(chan *CommandRequest, 256)
+	s.features = make(map[string]bool)
+	s.event = make(chan bool)
 	s.Name = identifier
 	s.Parent = parent
 	s.fileMonitor = monitor
 	tree.SlavesByName[identifier] = &s
 	return &s
 }
 
-// If the slave is executing, or has executed its action, trigger a restart.
-// There's no need to trigger a restart if the slave has not yet begun to
-// execute its action, and there's no need to queue multiple restarts.
 func (s *SlaveNode) RequestRestart() {
-	s.requestRestart(false)
-}
+	s.L.Lock()
```

Comment:
I mutate the `Error` field in this one whereas the other just writes to a channel.